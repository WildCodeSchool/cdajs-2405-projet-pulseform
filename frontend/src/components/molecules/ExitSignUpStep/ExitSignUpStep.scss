@use "@assets/variables" as *;

.exit-sign-up-step {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  height: 100vh;

  &__img {
    margin-bottom: clamp(1rem, 5vh, 3rem);
  }

  &__label {
    @include h1-phone;
    color: $primary;
    text-align: center;
    width: 80%;
  }
  &__label-first {
    @include h1-phone;
    color: $primary;
    text-align: center;
    width: 80%;
  }

  &__label-profile {
    @include h1-phone;
    color: $primary;
    text-align: center;
    margin: 1rem;
  }

  &__btn {
    margin-top: clamp(16px, 5vh, 2rem);
  }
}
