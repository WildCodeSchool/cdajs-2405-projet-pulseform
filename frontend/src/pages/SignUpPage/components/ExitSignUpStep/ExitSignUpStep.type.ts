export interface ExitSignUpStepProps {
	isCompleted?: boolean;
}
