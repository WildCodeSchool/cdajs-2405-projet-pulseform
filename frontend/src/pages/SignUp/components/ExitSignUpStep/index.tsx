import blobMenGirl from "@assets/images/blob-men-girl.svg";
import BasicButton from "@components/atoms/BasicButton";
import { useTranslation } from "react-i18next";
import "./ExitSignUpStep.scss";
import type { ExitSignUpStepProps } from "./ExitSignUpStep.type";

function ExitSignUpStep({ isCompleted = false }: ExitSignUpStepProps) {
  const { t } = useTranslation();

  return (
    <section className="exit-sign-up-step">
      <img
        className="exit-sign-up-step__img"
        src={blobMenGirl}
        alt="blob men girl"
      />
      <p className="exit-sign-up-step__label">{t("WELCOME")}</p>
      <p className="exit-sign-up-step__label-first">
        {t("START_YOUR_FIRST_PROGRAM")}
      </p>
      {!isCompleted && (
        <p className="exit-sign-up-step__label-profile">
          {t("COMPLITED_NEXT_TIME_PROFILE")}
        </p>
      )}
      <BasicButton className="exit-sign-up-step__btn">
        {t("DISCOVERY_PROGRAMS")}
      </BasicButton>
    </section>
  );
}

export default ExitSignUpStep;
