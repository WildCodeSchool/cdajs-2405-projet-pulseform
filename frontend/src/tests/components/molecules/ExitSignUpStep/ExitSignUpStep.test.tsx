import ExitSignUpStep from "@components/molecules/ExitSignUpStep";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

describe("ExitSignUpStep", () => {
  it("renders the ExitSignUpStep component", () => {
    const tree = render(
      <MemoryRouter>
        <ExitSignUpStep />
      </MemoryRouter>,
    );
    expect(tree).toMatchSnapshot();
  });

  it("renders the ExitSignUpStep component with is completed", () => {
    const tree = render(
      <MemoryRouter>
        <ExitSignUpStep isCompleted />
      </MemoryRouter>,
    );
    expect(tree).toMatchSnapshot();
  });
});
