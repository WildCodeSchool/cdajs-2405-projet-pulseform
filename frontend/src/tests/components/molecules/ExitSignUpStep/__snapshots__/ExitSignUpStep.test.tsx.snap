// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ExitSignUpStep > renders the ExitSignUpStep component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="exit-sign-up-step"
      >
        <img
          alt="blob men girl"
          class="exit-sign-up-step__img"
          src="/src/assets/images/blob-men-girl.svg"
        />
        <p
          class="exit-sign-up-step__label"
        >
          WELCOME
        </p>
        <p
          class="exit-sign-up-step__label-first"
        >
          START_YOUR_FIRST_PROGRAM
        </p>
        <p
          class="exit-sign-up-step__label-profile"
        >
          COMPLITED_NEXT_TIME_PROFILE
        </p>
        <button
          class="basic-button exit-sign-up-step__btn basic-button--basic"
          type="button"
        >
          DISCOVERY_PROGRAMS
        </button>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="exit-sign-up-step"
    >
      <img
        alt="blob men girl"
        class="exit-sign-up-step__img"
        src="/src/assets/images/blob-men-girl.svg"
      />
      <p
        class="exit-sign-up-step__label"
      >
        WELCOME
      </p>
      <p
        class="exit-sign-up-step__label-first"
      >
        START_YOUR_FIRST_PROGRAM
      </p>
      <p
        class="exit-sign-up-step__label-profile"
      >
        COMPLITED_NEXT_TIME_PROFILE
      </p>
      <button
        class="basic-button exit-sign-up-step__btn basic-button--basic"
        type="button"
      >
        DISCOVERY_PROGRAMS
      </button>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ExitSignUpStep > renders the ExitSignUpStep component with is completed 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="exit-sign-up-step"
      >
        <img
          alt="blob men girl"
          class="exit-sign-up-step__img"
          src="/src/assets/images/blob-men-girl.svg"
        />
        <p
          class="exit-sign-up-step__label"
        >
          WELCOME
        </p>
        <p
          class="exit-sign-up-step__label-first"
        >
          START_YOUR_FIRST_PROGRAM
        </p>
        <button
          class="basic-button exit-sign-up-step__btn basic-button--basic"
          type="button"
        >
          DISCOVERY_PROGRAMS
        </button>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="exit-sign-up-step"
    >
      <img
        alt="blob men girl"
        class="exit-sign-up-step__img"
        src="/src/assets/images/blob-men-girl.svg"
      />
      <p
        class="exit-sign-up-step__label"
      >
        WELCOME
      </p>
      <p
        class="exit-sign-up-step__label-first"
      >
        START_YOUR_FIRST_PROGRAM
      </p>
      <button
        class="basic-button exit-sign-up-step__btn basic-button--basic"
        type="button"
      >
        DISCOVERY_PROGRAMS
      </button>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
